File: Reqnroll/Chapter13/13.3-TheIntroduceAndReferPattern/Before/WIMP.IntroduceAndReferSample.App/Services/NotificationService.cs
using WIMP.IntroduceAndReferSample.App.Infrastructure;
using WIMP.IntroduceAndReferSample.App.Models;

namespace WIMP.IntroduceAndReferSample.App.Services;

public static class NotificationService
{
    public static void SendCancellationNotification(string customerName)
    {
        DataContext.Instance.SaveNotification(new Notification(customerName, "Your order has been cancelled."));
    }

    public static bool WasNotificationSent(string? customerName)
    {
        return DataContext.Instance.GetNotificationsByCustomerName(customerName ?? "n/a").Any();
    }
}

File: CucumberJVM/Chapter13/13.1-LocalScenarioContextPattern/Before/src/main/java/com/wimp/app/services/NotificationService.java
/*
 * NOTE: This application ("WIMP - Where Is My Pizza") is provided solely to
 * demonstrate the Behavior-Driven Development scenario automation patterns
 * described in the book "Effective Behavior-Driven Development" by
 * Gaspar Nagy and Seb Rose.
 *
 * It is NOT a complete or production-ready implementation. It deliberately
 * uses shortcuts and simplifications (e.g. authentication, data storage,
 * error handling, security) that are NOT suitable for a real application.
 * Do not use this code as a basis for production software.
 */

package com.wimp.app.services;

import com.wimp.app.infrastructure.DataRepository;
import com.wimp.app.models.Notification;

public class NotificationService {
    public static void sendCancellationNotification(String customerName) {
        DataRepository.INSTANCE.saveNotification(new Notification(customerName, "Your order has been cancelled."));
    }

    public static boolean wasNotificationSent(String customerName) {
        return !DataRepository.INSTANCE.getNotificationsByCustomerName(customerName).isEmpty();
    }
}
